src/main/java/com/faforever/client/leaderboard/LeaderboardService.java

```diff
@@ -102,26 +102,19 @@ public Flux<LeagueSeason> getActiveSeasons() {
   }
 
   @Cacheable(value = CacheNames.LEAGUE, sync = true)
-  public Mono<LeagueSeason> getLatestSeason(League league) {
+  public Flux<LeagueSeason> getSeasons(League league) {
     ElideNavigatorOnCollection<com.faforever.commons.api.dto.LeagueSeason> navigator = ElideNavigator.of(
                                                                                                          com.faforever.commons.api.dto.LeagueSeason.class)
                                                                                                      .collection()
                                                                                                      .setFilter(
                                                                                                          qBuilder().intNum(
                                                                                                                        "league.id")
-                                                                                                                   .eq(league.id())
-                                                                                                                   .and()
-                                                                                                                   .instant(
-                                                                                                                       "startDate")
-                                                                                                                   .before(
-                                                                                                                       OffsetDateTime.now()
-                                                                                                                                     .toInstant(),
-                                                                                                                       false))
+                                                                                                                   .eq(league.id()))
                                                                                                      .addSortingRule(
                                                                                                          "startDate",
                                                                                                          false);
-    return fafApiAccessor.getMany(navigator)
-                         .next().map(leaderboardMapper::map);
+    return fafApiAccessor.getMany(navigator).map(leaderboardMapper::map)
+                         .cache();
   }
```

Comment:
The only thing is that this list could get very long over time. So we may want some kind of pagination

Reply:
If we want to access a season that is not in the first page we would have to be able to load earlier pages. That sounds very cumbersome from a ui perspective. I'm not sure how we could reflect this pagination in the client ui

Reply:
besides, you are technically correct that this list could get long, but we have four seasons per year, so in ten years we will be at a total of around 50. I think this is still small enough to load it in one go

Reply:
It was less the loading I was thinking about and more how long the list in the UI could be and how that would be to navigate